### PR TITLE
Amp/is obstructed

### PR DESCRIPTION
**Pull request by amarkpark**
RESUBMITTED 20151203: This handles all cases (vectors) of piece movement. Horizontal pos & neg, vertical pos & neg, diag pos/pos, pos/neg, neg/pos, neg/neg. It tests most cases (of pass/fail in each of 8 vectors of movement). Previous PR was closed to avoid merging during refactor. Many revisions from original model method. Broke queries into 3 helper methods. Expanded refactored and reorganized tests. Proper failure (and test) on Invalid case.

### CODE REVIEW

**Comment by amarkpark** [comment]
I broke the functionality of the diagonal vectors in the refactor, it examines the same square multiple times. I need to fix that before merging.

**Comment by amarkpark** [comment]
Diagonal vector logic should be cycling through all interstitial squares again.

File: spec/models/piece_spec.rb [hunk 1]
```diff
@@ -105,4 +109,118 @@
       end
     end
   end
+
+  describe 'is_obstructed?' do
+    before :all do
+      @game = FactoryGirl.create(:game)
+      @game.pieces.delete
```
Comment (jeffgerlach) [inline spec/models/piece_spec.rb:116]: May need to use delete_all here instead of delete. Double-check with pry, will check when I get home

**Comment by amarkpark** [comment]
SCORE! Looks like "git rm --cached .byebug_history" did the trick.

**Comment by amarkpark** [comment]
Cleaner still, thanks to some pair-coding with @acodeinprogress! Changed helper methods from row and column queries to range queries that can have either coordinate held constant and full ranges in both directions if necessary.

**Pull request merged**